nx-X11/programs/Xserver/dix/window.c

```diff
@@ -2611,6 +2610,14 @@ ReparentWindow(register WindowPtr pWin, register WindowPtr pParent,
     /* insert at begining of pParent */
     pWin->parent = pParent;
     pPrev = RealChildHead(pParent);
+
+#ifdef NXAGENT_SERVER
+    if (pWin->parent == screenInfo.screens[0]->root)
+    {
+      nxagentSetTopLevelEventMask(pWin);
+    }
+#endif
+
```

Comment:
Copyright wise, this one is tricky. Can you move the if clause into some function in hw/nxagent/Utils.c or some such? And then simply call that little function in the DDX from here?